src/pages/workspace/withPolicy.tsx

```diff
@@ -61,61 +57,49 @@ const policyPropTypes = {
     policyMembers: PropTypes.objectOf(policyMemberPropType),
 };
 
-const policyDefaultProps = {
-    policy: {},
+type WithPolicyOnyxProps = {
+    policy: OnyxEntry<OnyxTypes.Policy>;
+    policyMembers: OnyxEntry<OnyxTypes.PolicyMember>;
+    policyDraft: OnyxEntry<OnyxTypes.Policy>;
+    policyMembersDraft: OnyxEntry<OnyxTypes.PolicyMember>;
+};
+
+type WithPolicyProps = WithPolicyOnyxProps & {
+    route: PolicyRoute;
+};
+
+const policyDefaultProps: WithPolicyOnyxProps = {
+    policy: {} as OnyxTypes.Policy,
     policyMembers: {},
+    policyDraft: {} as OnyxTypes.Policy,
+    policyMembersDraft: {},
 };
 
 /*
  * HOC for connecting a policy in Onyx corresponding to the policyID in route params
  */
-export default function (WrappedComponent) {
-    const propTypes = {
-        /** The HOC takes an optional ref as a prop and passes it as a ref to the wrapped component.
-         * That way, if a ref is passed to a component wrapped in the HOC, the ref is a reference to the wrapped component, not the HOC. */
-        forwardedRef: PropTypes.func,
+export default function <TProps extends WithPolicyProps, TRef>(WrappedComponent: ComponentType<TProps & RefAttributes<TRef>>): React.ComponentType<Omit<TProps, keyof WithPolicyOnyxProps>> {
+    function WithPolicy(props: TProps, ref: ForwardedRef<TRef>) {
+        const routes = useNavigationState((state) => state.routes || []);
```

Comment:
@blazejkustra why did we use `useNavigationState` here to get the current route instead of just using `useRoute`?

Reply:
I reused the logic that was written before, with Typescript migration I usually try to avoid refactoring that could cause regressions, also at the time navigation wasn't migrated yet so I just didn't want to touch it here 😄 
```ts
const currentRoute = _.last(useNavigationState((state) => state.routes || []));
```